### ISSUE TITLE
[feature] Switch cluster scaling to machine classes after creation

### ISSUE THREAD

**Issue by fracek** [siderolabs/omni#507]
### Problem Description

At the moment, it's not possible to switch cluster scaling to machine classes after creation. I receive the following error if I try: 

```
failed to validate: 1 error occurred: * machine set is not empty and is using manual nodes management, updating to machine class mode is not allowed
```

I expect to be able to use machine classes without re-creating the cluster. In my case, the set of machines in the machine class is the same set as the machines with the given role (CP, WO) so the operation is nearly a noop.

### Solution

Ideally, Omni should be smart enough to notice that the selected machine class contains the same nodes that currently have that role and switch to it.

**Comment by Unix4ever** [comment]
This is fixed in 0.40

**Comment by fracek** [comment]
My understanding is that the other issue is for the empty (default) machine set, my issue is for non empty sets that are a subset of the machine class set.
